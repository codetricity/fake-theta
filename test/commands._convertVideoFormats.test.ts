import {
  request,
  toMatchInvalidHeaderParameterError,
  toMatchMissingParameterError,
} from './helper';

describe('POST /osc/commands/execute _convertVideoFormats', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
        parameters: {
          fileUrl: 'http://192.168.1.1/files/100RICOH/R0010015.MP4',
          size: '3840 x 1920',
        },
      })
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
        parameters: {
          fileUrl: 'http://192.168.1.1/files/100RICOH/R0010015.MP4',
        },
      })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA X POST /osc/commands/execute _convertVideoFormats', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
        parameters: {
          fileUrl: 'http://192.168.1.1/files/100RICOH/R0010015.MP4',
          size: '3840 x 1920',
        },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
        parameters: {
          fileUrl: 'http://192.168.1.1/files/100RICOH/R0010015.MP4',
        },
      })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA Z1 POST /osc/commands/execute _convertVideoFormats', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
        parameters: {
          fileUrl:
            'http://192.168.1.1/files/150100525831424d42075b53ce68c300/100RICOH/R0010015.MP4',
          size: '3840 x 1920',
          projectionType: 'Equirectangular',
          codec: 'H.264/MPEG-4 AVC',
          topBottomCorrection: 'Apply',
        },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
        parameters: {
          fileUrl:
            'http://192.168.1.1/files/150100525831424d42075b53ce68c300/100RICOH/R0010015.MP4',
          size: '3840 x 1920',
        },
      })
      .set('emulating-theta-model', 'z1')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('Unknown Type POST /osc/commands/execute _convertVideoFormats', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._convertVideoFormats',
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
