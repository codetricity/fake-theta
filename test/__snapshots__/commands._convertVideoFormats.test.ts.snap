// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _convertVideoFormats should respond with results 1`] = `"{"id":"10","name":"camera._convertVideoFormats","state":"inProgress"}"`;

exports[`THETA X POST /osc/commands/execute _convertVideoFormats should respond with results 1`] = `"{"id":"10","name":"camera._convertVideoFormats","state":"inProgress"}"`;

exports[`THETA Z1 POST /osc/commands/execute _convertVideoFormats should respond with results 1`] = `"{"id":"10","name":"camera._convertVideoFormats","progress":{"completion":0},"state":"inProgress"}"`;
